components/ws-manager-api/typescript/src/client-provider-source.ts

```diff
@@ -27,8 +27,15 @@ export class WorkspaceManagerClientProviderEnvSource implements WorkspaceManager
         return this.clusters.find((m) => m.name === name && m.applicationCluster === applicationCluster);
     }
 
-    public async getAllWorkspaceClusters(applicationCluster: string): Promise<WorkspaceClusterWoTLS[]> {
-        return this.clusters.filter((m) => m.applicationCluster === applicationCluster) ?? [];
+    public async getAllWorkspaceClusters(
+        applicationCluster: string,
+        region?: string,
+    ): Promise<WorkspaceClusterWoTLS[]> {
+        return (
+            this.clusters.filter(
+                (m) => m.applicationCluster === applicationCluster && (!region || m.region === region),
```

Comment:
I think we'll want to remove the `applicationCluster` relationship here. This would only evaluate to true for the region-local cluster, which would prevent us from actually schedule it in EU, from the US.

Reply:
AFAIK the `applicationCluster` came in with the db-sync removal. In practice we'll have the same workspace cluster registered "multiple times" (when using one DB without db-sync), once for each application cluster.

Reply:
I would also like to remove it but not in this PR.